src/main/java/org/openstreetmap/atlas/generator/sharding/OverpassClient.java

```diff
@@ -164,8 +180,9 @@ public CloseableHttpResponse getResponse(final String specificQuery)
         try (CloseableHttpResponse response = this.getResponse(query))
         {
             logger.info("Parsing node query response...");
-            final Document doc = builder.parse(response.getEntity().getContent());
-            final NodeList nodelist = doc.getElementsByTagName("node");
+            final Document document = builder.parse(response.getEntity().getContent());
+            checkForTooMuchData(document);
+            final NodeList nodelist = document.getElementsByTagName("node");
```

Comment:
If we see too much data returned, does it even make sense to proceed with the code below? What are your thoughts on bailing out early - maybe throwing Exception or returning empty list.